api/src/main/java/org/openmrs/module/reporting/config/ReportDescriptor.java

```diff
@@ -0,0 +1,109 @@
+package org.openmrs.module.reporting.config;
+
+import org.codehaus.jackson.annotate.JsonProperty;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+public class ReportDescriptor {
+
+    @JsonProperty
+    private String key;
+
+    @JsonProperty
+    private String uuid;
+
+    @JsonProperty
+    private String name;
+
+    @JsonProperty
+    private String description;
+
+    @JsonProperty
+    private List<ParameterDescriptor> parameters;
+
+    @JsonProperty
+    private List<DataSetDescriptor> datasets;
+
+    @JsonProperty
+    private List<DesignDescriptor> designs;
+
+    @JsonProperty
+    private Map<String, Object> config;
+
+    // utility property for storing the path to this descriptor on the file system so we know where to look for related files
+    private File path;
```

Comment:
FYI... this smelled a bit to me, but seemed the simplest way to do it... (this is to allow the search for sql files and report templates to be relative to the directory the yml file is in without having to pass the path around everywhere)